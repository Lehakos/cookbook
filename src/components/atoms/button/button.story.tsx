import React from 'react';
import { storiesOf } from '@storybook/react';
import { select, boolean } from '@storybook/addon-knobs';

import { Button } from './button';

const stories = storiesOf('Atoms|Button', module);

stories.add('Default', () => (
  <Button
    disabled={boolean('disabled', false)}
    variant={select('variant', ['primary', 'secondary'], 'primary') as any}
  >
    Click me
  </Button>
));
