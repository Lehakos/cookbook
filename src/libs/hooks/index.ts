export { useControlledProp } from './use-controlled-prop';
export { useElementSize } from './use-element-size';
