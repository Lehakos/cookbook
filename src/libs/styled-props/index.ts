export { typography, TypographyProps } from './typography';
export { layout, LayoutProps } from './layout';
export { flexbox, FlexboxProps } from './flexbox';
