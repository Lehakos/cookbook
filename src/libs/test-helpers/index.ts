export { render } from './render';
export { mockElementWithBoundingRect } from './mock-element-with-bounding-rect';
