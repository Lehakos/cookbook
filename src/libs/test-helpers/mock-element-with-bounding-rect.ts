export const mockElementWithBoundingRect = ({
  width = 100,
  height = 100,
  top = 0,
  left = 0,
}) => {
  const right = left + width;
  const bottom = top + height;

  Object.defineProperties(Element.prototype, {
    offsetLeft: {
      get() {
        return left;
      },
      configurable: true,
    },
    offsetTop: {
      get() {
        return top;
      },
      configurable: true,
    },
    offsetHeight: {
      get() {
        return height;
      },
      configurable: true,
    },
    offsetWidth: {
      get() {
        return width;
      },
      configurable: true,
    },
  });

  Element.prototype.getBoundingClientRect = jest.fn(() => {
    return {
      width,
      height,
      top,
      left,
      bottom,
      right,
    };
  });

  const element = document.createElement('div');

  return element;
};
