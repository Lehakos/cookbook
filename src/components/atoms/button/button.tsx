import React from 'react';

import * as s from './button.styles';

interface Props {
  /**
   * Design of button
   * @default 'primary'
   */
  variant?: s.Variant;
  disabled?: boolean;
}

export const Button: React.FunctionComponent<Props> = function({
  children,
  disabled,
  variant = 'primary',
  ...rest
}) {
  return (
    <s.Wrapper {...rest} disabled={disabled} variant={variant}>
      {children}
    </s.Wrapper>
  );
};
