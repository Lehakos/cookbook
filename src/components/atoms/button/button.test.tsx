import React from 'react';

import { render } from 'libs/test-helpers';

import { Button } from './button';

describe('<Button />', () => {
  it('renders properly', () => {
    const { container } = render(<Button />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
