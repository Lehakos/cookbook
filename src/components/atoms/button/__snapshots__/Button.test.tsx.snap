// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders properly 1`] = `
.c0 {
  padding: 12px 24px;
  border: none;
  color: #fff;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.05em;
  -moz-letter-spacing: 0.05em;
  -ms-letter-spacing: 0.05em;
  letter-spacing: 0.05em;
  border-radius: 4px;
  cursor: pointer;
  background: #F47983;
  -webkit-transition: background-color 0.2s;
  transition: background-color 0.2s;
}

.c0[disabled] {
  cursor: default;
}

.c0:hover {
  background-color: #f3707a;
}

.c0:active {
  background-color: #f2616d;
}

.c0[disabled] {
  background-color: #f79aa2;
}

<button
  class="c0"
/>
`;
